cpp/velox/operators/functions/SparkPartitionId.cc

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "SparkPartitionId.h"
+#include "config/GlutenConfig.h"
+
+#include "velox/expression/VectorFunction.h"
+#include "velox/type/Type.h"
+
+namespace gluten {
+
+class SparkPartitionId final : public facebook::velox::exec::VectorFunction {
```

Comment:
Just wondering why we prefer to implement this function in Gluten. Is there any chance to upstream it in Velox?

Reply:
I thought this function is tightly bound to gluten. And you see the logic of this function is very simple. The only meaning to support this is to avoid fallback because of this function.

Reply:
I can split this to two changes and add it in upstream velox, if you prefer this way, thanks.

Reply:
It seems a bit strange to me. The function just returns the value stored in config.  Do we really need to implement a native version of spark_partition_id?

Reply:
We observed the fallback with this function, as the implement is very simple, so I think we should avoid this fallback.
Please let me know if you have other suggestion, like we should handle this in gluten side, etc.

Reply:
Yes, let's try to upstream to Velox firstly.

Reply:
Sure, get it.